chatlearn/utils/vllm_utils.py

```diff
@@ -736,22 +736,26 @@ def get_megatron_sharded_states(args, tp_size, pp_size, pp_rank):
     return tp_state_dicts
 
 
-def load_state_dict(args):
-    # Load original state dict from Megatron-LM checkpoint.
+def load_rank0_state_dict(args):
+    # Load rank0 state dict from Megatron-LM checkpoint.
     possible_sub_dirs = ["mp_rank_00", "mp_rank_00_000"]
 
     for root, dirnames, _ in os.walk(args["load"]):
         for dirname in dirnames:
-            if dirname in possible_sub_dirs:
-                sub_dir = os.path.join(root, dirname)
-                if os.path.exists(sub_dir):
-                    rank0_checkpoint_name = glob.glob(sub_dir + "/model*.pt")
-                    args["load"] = root
-                    rank0_checkpoint_path = rank0_checkpoint_name[0]
+            if dirname not in possible_sub_dirs:
+                continue
+            sub_dir = os.path.join(root, dirname)
+            if not os.path.exists(sub_dir):
+                continue
+            rank0_checkpoint_name = glob.glob(sub_dir + "/model*.pt")
+            if len(rank0_checkpoint_name) == 0:
+                continue
+            args["load"] = root
+            rank0_checkpoint_path = rank0_checkpoint_name[0]
```

Comment:
whether needed to check rank0_checkpoint_path is None?

Reply:
`glob.glob` always returns a possibly-empty list of path names. It won't be None unless the api is changed.